cli/module_registry.go

```diff
@@ -637,3 +645,36 @@ func getEntrypointForVersion(mod *apppb.Module, version string) (string, error)
 	// if there is no entrypoint set yet, use the last uploaded entrypoint
 	return mod.Entrypoint, nil
 }
+
+func isTarball(path string) bool {
+	return strings.HasSuffix(strings.ToLower(path), ".tar.gz") ||
+		!strings.HasSuffix(strings.ToLower(path), ".tgz")
+}
```

Comment:
!strings.HasSuffix(strings.ToLower(file.Name()), ".tar.gz") &&
		!strings.HasSuffix(strings.ToLower(file.Name()), ".tgz")

Reply:
> to validate integrity

What could you validate about the created tarball? That it was `tar`able?

Reply:
Also why are you checking `!strings.HasSuffix(strings.ToLower(path), ".tgz")`? Won't this return `true` for most files?